backends-velox/src/main/scala/org/apache/gluten/execution/ColumnarCollectLimitExec.scala

```diff
@@ -125,11 +199,20 @@ case class ColumnarCollectLimitExec(
       if (childRDD.getNumPartitions == 1) childRDD
       else shuffleLimitedPartitions(childRDD)
 
-    processedRDD.mapPartitions(partition => collectLimitedRows(partition, limit))
+    processedRDD.mapPartitions(
+      partition => {
+        val droppedRows = dropLimitedRows(partition, offset)
+        val adjustedLimit = Math.max(0, limit - offset)
+        collectLimitedRows(droppedRows, adjustedLimit)
```

Comment:
Can we enhance the collectLimitedRows, we can slice the input RowVector from offset to adjustedLimit?